staging/src/k8s.io/cluster-bootstrap/util/secrets/secrets.go

```diff
@@ -45,8 +45,8 @@ func GetData(secret *v1.Secret, key string) string {
 }
 
 // HasExpired will identify whether the secret expires
-func HasExpired(secret *v1.Secret, currentTime time.Time) bool {
-	_, expired := GetExpiration(secret, currentTime)
+func HasExpired(logger klog.Logger, secret *v1.Secret, currentTime time.Time) bool {
```

Comment:
I believe we should use a module-level logger here instead of wiring it this way.

Reply:
Hmmmm, I see that that's not a pattern that kubernetes follows. That's a shame. You can ignore the above comment.

Reply:
I appreciate your comment.
We can pass ctx instead of logger.
e.g.
```
func HasExpired(ctx context.Context, secret *v1.Secret, currentTime time.Time) bool {}
```
Do you prefer this?

Reply:
I think it's ok to just use what you've got. I originally came from python where you'd set up a logger per-module so that you don't have to pass it in arguments. It's possible that that pattern would be harder to implement in golang given it's a compiled language.